internal/metrics/metrics.go

```diff
@@ -0,0 +1,38 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	ctrlmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	AddonsPerPhaseTotal = prometheus.NewGaugeVec(
```

Comment:
All the metrics in this PR rely on Phase. And since it took >1month to get this PR reviewed rebasing against main is a real pain. I feel like just closing this PR.

WDYT?

Reply:
:shrug:  I can take it over if you want.

Reply:
That would be awesome. I don't think I have the bandwidth to fix this for the current sprint. Thank you so much.